libs/filameshio/src/MeshReader.cpp

```diff
@@ -159,30 +159,32 @@ static size_t fileSize(int fd) {
 
 namespace filamesh {
 
-MeshReader::Mesh MeshReader::loadMeshFromFile(filament::Engine* engine, const utils::Path& path,
-        MaterialRegistry& materials) {
-
+MeshReader::Mesh MeshReader::loadMeshFromFile(filament::Engine *const engine, const utils::Path& path,
+        MaterialRegistry& materialRegistry) {
     Mesh mesh;
 
-    int fd = open(path.c_str(), O_RDONLY);
+    /* Open the mesh file for reading only. */
+    const int fd = open(path.c_str(), O_RDONLY);
+
+    /* Check if an error occurred when opening the file. */
+    if (fd == -1) return mesh;
 
-    size_t size = fileSize(fd);
-    char* data = (char*) malloc(size);
-    read(fd, data, size);
+    const size_t size = fileSize(fd);
+    const char *const data = (char *)malloc(size);
 
     if (data) {
-        char *p = data;
-        char *magic = p;
-        p += sizeof(MAGICID);
+        (void)read(fd, (void *)data, size);
 
-        if (!strncmp(MAGICID, magic, 8)) {
-            mesh = loadMeshFromBuffer(engine, data, nullptr, nullptr, materials);
+        if (strncmp(MAGICID, data, 8) == 0) {
+            mesh = loadMeshFromBuffer(engine, data, nullptr, nullptr, materialRegistry);
         }
 
         Fence::waitAndDestroy(engine->createFence());
-        free(data);
+        free((void *)data);
     }
-    close(fd);
+
+    /* Close the file. */
+    (void)close(fd);
```

Comment:
don't do that.

Reply:
Why?

Reply:
it's not a style we're using anywhere else in filament. I don't want to start diverging in style, without a strong benefit.

Reply:
OK, but the style regarding this type of sentence is not clear.